tests/testsuite/shell_quoting.rs

```diff
@@ -26,14 +25,24 @@ fn features_are_quoted() {
         .build();
 
     p.cargo("check -v")
-            .env("MSYSTEM", "1")
-            .with_status(101)
-            .with_stderr_contains(
-                r#"[RUNNING] `rustc [..] --cfg 'feature="default"' --cfg 'feature="some_feature"' [..]`"#
-            ).with_stderr_contains(
-                r#"
+        .env("MSYSTEM", "1")
+        .with_status(101)
+        .with_stderr_data(str![])
+        .with_stderr_data(str![[r#"
+[CHECKING] foo v0.1.0 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo --edition=2015 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,metadata -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="some_feature"' -C metadata=c2925e00c1458fcb -C extra-filename=-c2925e00c1458fcb --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps`
+error[E0425]: cannot find value `error` in this scope
+ --> src/main.rs:1:12
+  |
+1 | fn main() {error}
+  |            ^^^^^ not found in this scope
+
+For more information about this error, try `rustc --explain E0425`.
```

Comment:
Snapshotting rustc output is a bit risky. If applicable we should omit it by `...`.

```suggestion
...
```

Reply:
By risky I mean rustc output is not guaranteed stable and might break cargo test suites between different toolchain versions.

Reply:
I've added this to the porting steps

> - If replacing a `contains` with an equality check, use multi-line globs (`...`) for rustc errors that weren't previously matched to minimize tying Cargo's tests to the exact output of rustc